packages/blade/src/components/Input/BaseInput/StyledBaseInput.native.tsx

```diff
@@ -113,11 +133,32 @@ export const StyledBaseInput = React.forwardRef<TextInput, StyledBaseInputProps>
       type,
       numberOfLines,
       isTextArea,
+      hasPopup,
       ...props
     },
     ref,
   ) => {
-    return (
+    return hasPopup ? (
+      <StyledNativeBaseButton
+        // the types of styled-components for react-native is creating a mess, so there's no other option but to type `ref` as any
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        ref={ref as any}
+        // @ts-expect-error: not able to type onClick for both web and native
```

Comment:
Lets say I type onClick for React Native, then we start getting typescript error on web on it's parent elements and if we type it for web, we get errors on RN.

I didn't spend much time here because there isn't any code that we write here and it is anyway typed for web on `BaseInput` so we still get autocompletion outside.

Reply:
Update: Fixed - https://github.com/razorpay/blade/pull/932#discussion_r1085081773